examples/credential.rs

```diff
@@ -53,13 +54,13 @@ async fn main() -> Result<()> {
   // Create an unsigned Credential with claims about `subject` specified by `issuer`.
   let credential: Credential = issue_degree(&doc_iss, &doc_sub)?;
 
-  // Sign the Credential with the issuer secret key - the result is a VerifiableCredential.
-  let credential: VerifiableCredential = credential.sign(&doc_iss, "#authentication".into(), key_iss.secret())?;
+  // Sign the Credential with the issuer secret key - the result is a Credential.
+  let credential: Credential = credential.sign(&doc_iss, "#authentication".into(), key_iss.secret())?;
```

Comment:
This is also possible now:

```rust
doc_iss.sign_data(&mut credential, key_iss.secret())?;
```

I think it makes sense to drop the `sign` function from the actual `Credential` type now. All types that implement `Serialize` and `SetSignature` can be signed through the `sign_data` function (this implicitly uses the authentication method)

Reply:
Should be able to use the `DocumentSigner`:

```rust
document.signer(keypair.secret()).method("#key-1").sign(&mut credential)?;
```